src/dictionary/methods/listeners.hpp

```diff
@@ -0,0 +1,77 @@
+//
+// Created by Cesar Valdez on 07/03/2021.
+//
+
+#ifndef REALMJS_LISTENERS_HPP
+#define REALMJS_LISTENERS_HPP
+
+#include "dictionary/methods/callbacks.hpp"
+#include "napi.h"
+namespace realm {
+namespace js {
+
+template <typename VM>
+class ListenersMethodsForDictionary {
+   private:
+    using Value = js::Value<VM>;
+
+    template <typename Fn, typename JavascriptPlainObject>
+    auto add_javascript_function(std::string&& name, Fn&& function,
+                                 JavascriptPlainObject* object) {
+        auto plain_object = object->get_plain_object();
+        auto context = object->get_context();
+
+        auto fn = Napi::Function::New(context, function, name,
+                                      static_cast<void*>(object));
+        js::Object<VM>::set_property(context, plain_object, name, fn,
+                                     PropertyAttributes::DontEnum);
+    }
+
+   public:
+    template <typename JavascriptPlainObject>
+    void apply(JavascriptPlainObject* object) {
+        add_javascript_function("addListener", add_listener(object), object);
+        add_javascript_function("removeListener", remove_listener(object),
+                                object);
+        add_javascript_function("removeAllListeners",
+                                remove_all_listeners(object), object);
+    }
+
+    template <typename JavascriptPlainObject>
+    auto add_listener(JavascriptPlainObject* object) {
+        return [=](const Napi::CallbackInfo& info) {
```

Comment:
By using `js::Function<VM>` you can support any engine.

Reply:
I would like to do it, but that [Napi::CallbackInfo](https://github.com/nodejs/node-addon-api/blob/main/doc/callbackinfo.md) is not a function, is some kind of object injected into functions. I would suggest waiting for the port to JSC and we re-assess.